apps/docs/content/guides/auth/social-login/auth-google.mdx

```diff
@@ -376,81 +376,74 @@ If you're integrating Google One-Tap with your Next.js application, you can refe
 
 import Script from 'next/script'
 import { createClient } from '@/utils/supabase/client'
-import { CredentialResponse } from 'google-one-tap'
+import type { accounts, CredentialResponse } from 'google-one-tap'
 import { useRouter } from 'next/navigation'
 import { useEffect } from 'react'
 
+declare const google: { accounts: accounts }
+
+// generate nonce to use for google id token sign-in
+const generateNonce = async (): Promise<string[]> => {
+  const nonce = btoa(String.fromCharCode(...crypto.getRandomValues(new Uint8Array(32))))
+  const encoder = new TextEncoder()
+  const encodedNonce = encoder.encode(nonce)
+  const hashBuffer = await crypto.subtle.digest('SHA-256', encodedNonce)
+  const hashArray = Array.from(new Uint8Array(hashBuffer))
+  const hashedNonce = hashArray.map((b) => b.toString(16).padStart(2, '0')).join('')
+
+  return [nonce, hashedNonce]
+}
```

Comment:
Moved this out of the component since it has no dependency on the component itself